src/gui/EffectView.cpp

```diff
@@ -208,6 +214,42 @@ void EffectView::contextMenuEvent( QContextMenuEvent * )
 
 
 
+void EffectView::mousePressEvent(QMouseEvent* event)
+{
+	if (event->button() == Qt::LeftButton)
+	{
+		m_dragging = true;
+		m_opacityEffect->setOpacity(0.3);
+		update();
+	}
+}
+
+void EffectView::mouseReleaseEvent(QMouseEvent* event)
+{
+	if (event->button() == Qt::LeftButton)
+	{
+		m_dragging = false;
+		m_opacityEffect->setOpacity(1);
+		update();
+	}
+}
+
+void EffectView::mouseMoveEvent(QMouseEvent* event)
+{
+	if (m_dragging)
+	{
+		if (event->pos().y() < 0)
+		{
+			moveUp();
+		}
+		else if (event->pos().y() > EffectView::DEFAULT_HEIGHT)
+		{
+			moveDown();
+		}
+	}
```

Comment:
We can reduce indention by inverting the if guard. :)
```suggestion
	if (!m_dragging) { return; }
	if (event->pos().y() < 0)
	{
		moveUp();
	}
	else if (event->pos().y() > EffectView::DEFAULT_HEIGHT)
	{
		moveDown();
	}
```